text/0117-mobile-transactions-and-spans.md

```diff
@@ -0,0 +1,67 @@
+
+3. Automatic vs. manual transactions: Any user-created transaction potentially breaks the behavior of the existing automatic-transaction generation, since there can only be a single transaction on the scope.
+
+4. Transaction durations are sometimes pointless: As they’re based on idle transaction. Any relevant or even non-relevant child span can influence the transaction duration.
+
+# Options Considered
+
+### 1. Transactions as Carriers for Spans (preferred approach)
+
+The idea is to use transactions as carriers for measurements on mobile instead of trying to make the backend heavy concept work on mobile. There is no active "open" transaction, but we rather create transactions on-demand, whenever there is span data to send.
+In Sentry, we need pick apart these special transactions (that we shouldn’t consider as such anymore, maybe use `transaction->transaction_info->source` to identify carrier transaction) and only use their content, which are the spans we measured.
+
+TBD:
+* How can we ensure that spans are packaged up efficiently, ultimately creating not to little and not too many web requests.
+* As of now, some performance grouping is done based on transactions op and description. With this change they will simply act as carieers, so we need to ensure the span context has enough information that aggregation (e.g. by screen) can still be performed.
+* Profiles are right now captured based on transaction start signals, we need to find a reliable way to keep the remaining functionality.
+
+
+**Pros**
+
+- We always get what we want when we want it
+- Once we can ingest spans, it’ll be easy to switch.
+
+**Cons**
+
+- We need to change how we interact with transaction in the SDK and in the product
+
+### 2. One Transaction per Screen
+
+Have one transaction per screen. This automatic transaction can be fully managed within the Sentry SDK. If no spans are generated an empty transaction would still be sent, as it contains slow/frozen frame metrics and ttid/ttfd data.
+
+**Pros**
+
+- There’s always a transaction running, which spans can be attached to
+- There is no need for idle transactions anymore, as the lifetime matches a screen lifetime (a max deadline maybe still makes sense), making the overall behavior more predictable
+- User interactions are modeled as spans instead of transactions, so they don’t interfere with transactions
+- Slow and frozen frames could be added as child spans to the running transaction
+
+**Cons**
+
+- The transaction duration still makes no sense, as it’s length is determined by the time the screen was visible. Also probably needs a max timeout again too
+- Any manually added / background work spans (e.g. a long running sync) could extend the transaction lifetime, which could result in overlapping transactions
+- We need new (manual) APIs to track the current screen
```

Comment:
This won't work with multiple screens. Furthermore, this most likely won't work properly for declarative UI frameworks such as Compose or SwiftUI, where identifying the current screen is a challenge.